packages/server/lib/browsers/webkit-automation.ts

```diff
@@ -74,16 +80,89 @@ const _cookieMatches = (cookie: any, filter: Record<string, any>) => {
   return true
 }
 
+let requestIdCounter = 1
+const requestIdMap = new WeakMap<playwright.Request, string>()
+
 export class WebkitAutomation {
-  private context: playwright.BrowserContext
-  public reset: (url?: string) => Promise<void>
-
-  constructor (resetPage, private page: playwright.Page) {
-    this.context = page.context()
-    this.reset = async (url?: string) => {
-      this.page = await resetPage(url)
-      this.context = this.page.context()
-    }
+  private context!: playwright.BrowserContext
+  private page!: playwright.Page
+
+  private constructor (public automation: Automation, private browser: playwright.Browser) {}
+
+  // static initializer to avoid "not definitively declared"
+  static async create (automation: Automation, browser: playwright.Browser, initialUrl: string) {
+    const wkAutomation = new WebkitAutomation(automation, browser)
+
+    await wkAutomation.reset(initialUrl)
+
+    return wkAutomation
+  }
+
+  public async reset (newUrl?: string) {
+    debug('resetting playwright page + context %o', { newUrl })
+    // new context comes with new cache + storage
+    const newContext = await this.browser.newContext({
+      ignoreHTTPSErrors: true,
+    })
+    const oldPwPage = this.page
+
+    this.page = await newContext.newPage()
+    this.context = this.page.context()
+
+    this.attachListeners(this.page)
+
+    let promises: Promise<any>[] = []
+
+    if (oldPwPage) promises.push(oldPwPage.context().close())
+
+    if (newUrl) promises.push(this.page.goto(newUrl))
+
+    if (promises.length) await Promise.all(promises)
+  }
+
+  private attachListeners (page: playwright.Page) {
+    // emit preRequest to proxy
+    page.on('request', async (request) => {
```

Comment:
nit: no need for async, doesn't look like we're awaiting anything here yet

Reply:
```suggestion
    page.on('request', (request) => {
```